launcher/container_runner.go

```diff
@@ -552,13 +560,34 @@ func (r *ContainerRunner) Run(ctx context.Context) error {
 		return fmt.Errorf("failed to measure CEL events: %v", err)
 	}
 
-	if err := r.fetchAndWriteToken(ctx); err != nil {
-		return fmt.Errorf("failed to fetch and write OIDC token: %v", err)
+	// Only refresh token if agent has a default GCA client (not ITA use case).
+	if r.launchSpec.ITARegion == "" {
```

Comment:
This feels odd. Maybe instead we give each verifier a "auto refresh" method that we call into here? For ITA we can pass a method that does nothing and give this one to the GCA method. WDYT?

Reply:
Hmm IMO that seems complicated. If we're not refreshing the ITA token it's simpler to not deal with a coroutine at all and disable all that logic at the top level (container runner). The code is also more readable in this case (ITA = no refresh), despite the awkward conditional check.

Would it be better if we change the conditional to a nil check for the GCA client? Or any suitable check related to GCA rather than ITA?

Reply:
Resolved offline